x-pack/elastic-agent/pkg/composable/providers/kubernetes/config.go

```diff
@@ -9,24 +9,62 @@ package kubernetes
 
 import (
 	"time"
+
+	"github.com/elastic/beats/v7/libbeat/logp"
 )
 
 // Config for kubernetes provider
 type Config struct {
+	Scope     string    `config:"scope"`
+	Resources Resources `config:"resources"`
+
 	KubeConfig     string        `config:"kube_config"`
+	Namespace      string        `config:"namespace"`
 	SyncPeriod     time.Duration `config:"sync_period"`
 	CleanupTimeout time.Duration `config:"cleanup_timeout" validate:"positive"`
 
-	// Needed when resource is a pod
+	// Needed when resource is a Pod or Node
 	Node string `config:"node"`
 
-	// Scope of the provider (cluster or node)
-	Scope string `config:"scope"`
+	LabelsDedot      bool `config:"labels.dedot"`
+	AnnotationsDedot bool `config:"annotations.dedot"`
+}
+
+// Resources config section for resources' config blocks
+type Resources struct {
+	Pod     Enabled `config:"pod"`
+	Node    Enabled `config:"node"`
+	Service Enabled `config:"service"`
+}
+
+// Enabled config section for resources' config blocks
+type Enabled struct {
+	Enabled bool `config:"enabled"`
 }
 
 // InitDefaults initializes the default values for the config.
 func (c *Config) InitDefaults() {
-	c.SyncPeriod = 10 * time.Minute
 	c.CleanupTimeout = 60 * time.Second
```

Comment:
Yeap,  the provider is not really aware of the inputs right now, but maybe it could be handled better in the future if we introduce a "smart" controller which enables providers according to the inputs.